matrix/PseudoInverse.hpp

```diff
@@ -21,31 +21,38 @@ namespace matrix
  * Courrieu, P. (2008). Fast Computation of Moore-Penrose Inverse Matrices, 8(2), 25–29. http://arxiv.org/abs/0804.4809
  */
 template<typename Type, size_t M, size_t N>
-Matrix<Type, N, M> geninv(const Matrix<Type, M, N> & G)
+bool geninv(const Matrix<Type, M, N> & G, Matrix<Type, N, M>& res)
 {
     size_t rank;
     if (M <= N) {
         SquareMatrix<Type, M> A = G * G.transpose();
         SquareMatrix<Type, M> L = fullRankCholesky(A, rank);
 
-        SquareMatrix<Type, M> LtL = L.transpose() * L;
+        A = L.transpose() * L;
```

Comment:
So, this renaming shouldn't actually affect anything performance wise (and IMO makea the algorithm harder to understand), since the first thing the compiler does anyway is transform the code back into a SSA representation anyways.

Reply:
Interesting, maybe there were some side-effects in the assignment constructor or the stack address offsets from the extra structure are somehow important.